viabel/approximations.py

```diff
@@ -547,4 +547,127 @@ def _pth_moment(self, var_param, p):
 
     def supports_pth_moment(self, p):
         return False
+    
+def _get_low_rank_mu_sigma_pattern(dim, k):
+    ms_pattern = PatternDict(free_default=True)
+    ms_pattern['mu'] = NumericVectorPattern(length=dim)
+    ms_pattern['log_sigma'] = NumericVectorPattern(length=dim)
+    ms_pattern['low_rank'] = NumericArrayPattern(shape=(dim, k))
+    return ms_pattern
+
+class LRGaussian(ApproximationFamily):
+    """A low rank Gaussian approximation family."""
+
+    def __init__(self, dim, seed=1, k=0):
+        """Create multivariate Gaussian approximation family.
+
+        Parameters
+        ----------
+        dim : `int`
+            dimension of the underlying parameter space
+        k : 'int'
+            number of low rank
+        """
+        self._rs = npr.RandomState(seed)
+        self._pattern = _get_low_rank_mu_sigma_pattern(dim, k)
+        self._k = k
+        super().__init__(dim, self._pattern.flat_length(True), True, True)
+
+    def init_param(self):
+        init_param_dict = dict(mu=np.zeros(self.dim),
+                               log_sigma=np.ones(self.dim),
+                               low_rank=self._rs.randn(self.dim,self._k))
+        return self._pattern.flatten(init_param_dict)
+
+
+
+    def sample(self, var_param, n_samples, seed=None):
+        my_rs = self._rs if seed is None else npr.RandomState(seed)
+        param_dict = self._pattern.fold(var_param)
+        z = my_rs.randn(n_samples,self._k)
+        epsilon = my_rs.randn(n_samples, self.dim)
+        D_exp = np.exp(param_dict['log_sigma'])
+        B = param_dict['low_rank']
+
+        return param_dict['mu'] + np.dot(z,B.T) + D_exp * epsilon
+
+    def _entropy(self, var_param):
+        param_dict = self._pattern.fold(var_param)
+        B = param_dict['low_rank']
+        D_exp = np.exp(2 * param_dict['log_sigma'])
+        Sigma = np.diag(D_exp) + np.dot(B, B.T)
+        Sigma_det = np.linalg.det(Sigma)
+        return 0.5 * self.dim * ( np.log(2 * np.pi * np.e)) + 0.5 * np.log(Sigma_det)
 
+    def _kl(self, var_param0, var_param1):
+        param_dict0 = self._pattern.fold(var_param0)
+        param_dict1 = self._pattern.fold(var_param1)
+        mean_diff = param_dict0['mu'] - param_dict1['mu']
+
+        B0 = param_dict0['low_rank']
+        D0 = np.exp(2 * param_dict0['log_sigma'])
+        Sigma0 = np.diag(D0) + np.dot(B0, B0.T)
+        Sigma0_det = np.linalg.det(Sigma0)
```

Comment:
same comment as in the entropy case about efficiently computing the determinant. Also, since this is done in three different places, you should create a separate helper function to do this